nomenclature/testing.py

```diff
@@ -61,6 +61,11 @@ def assert_valid_structure(
         )
     if dimensions is None:
         dimensions = [x.stem for x in (path / definitions).iterdir() if x.is_dir()]
+        if not dimensions:
+            raise FileNotFoundError(
+                f"`Definitions` directory is empty : {path / definitions}"
+                '. Specify --dimensions "[]" if this is intended.'
```

Comment:
Not sure about this suggestion. 
If `"[]"` is provided, `DataStructureDefinition` is empty and as a result mappings cannot be validated either so `assert_valid_structure` would be pointless. If there are no definitions, you should not run `assert_valid_structure` at all.
I think I would just leave the suggestion out. 
```suggestion
```

Reply:
True...

Reply:
@phackstock Regarding this comment, this would actually mean that if people run validate-project with `--dimensions "[]"`, it would be useless. Should we somehow forbid this command?

Reply:
I could change the code so that if we type `--dimensions "[]"` it still looks for all the available dimensions...

Reply:
Agree with @phackstock, it should not be possible to validate with an empty definitions folder, so --dimensions "[]" should also raise an error. See my inline suggestion.